packages/cardano-services/src/Asset/DbSyncAssetProvider.ts

```diff
@@ -0,0 +1,131 @@
+import * as Queries from './queries';
+import {
+  Asset,
+  AssetProvider,
+  Cardano,
+  HealthCheckResponse,
+  Provider,
+  ProviderError,
+  ProviderFailure
+} from '@cardano-sdk/core';
+import {
+  CardanoTokenRegistry,
+  CardanoTokenRegistryConfiguration,
+  CardanoTokenRegistryDependencies
+} from './CardanoTokenRegistry';
+import { DbSyncNftMetadataService, DbSyncNftMetadataServiceDependencies } from './DbSyncNftMetadataService';
+import { NftMetadataService, TokenMetadataService } from './types';
+import { QueryResult } from 'pg';
+import { createDbSyncMetadataService } from '../Metadata';
+
+/**
+ * Configuration options for DbSyncAssetProvider
+ */
+export type DbSyncAssetProviderConfiguration = CardanoTokenRegistryConfiguration;
+
+/**
+ * Dependencies that are need to create DbSyncAssetProvider
+ */
+export type DbSyncAssetProviderDependencies = {
+  /**
+   * The NftMetadataService to retrieve Asset.NftMetadata .
+   * Default: new DbSyncNftMetadataService with dependencies taken from DbSyncAssetProviderDependencies
+   */
+  ntfMetadataService?: NftMetadataService;
+
+  /**
+   * The TokenMetadataService toretrieve Asset.TokenMetadata .
+   * Default: new CardanoTokenRegistry with dependencies taken from DbSyncAssetProviderDependencies
+   * and configuration from DbSyncAssetProviderConfiguration
+   */
+  tokenMetadataService?: TokenMetadataService;
+} & CardanoTokenRegistryDependencies &
+  Pick<DbSyncNftMetadataServiceDependencies, 'logger' | 'db'> &
+  Partial<Pick<DbSyncNftMetadataServiceDependencies, 'metadataService'>>;
+
+interface DbSyncAssetProviderDependenciesWithRequired extends DbSyncAssetProviderDependencies {
```

Comment:
I think we can separate the concerns here and just make the two services required properties of `DbSyncAssetProviderDependencies`. This would decouple the `DbSyncAssetProvider` from concrete implementations, and remove the type complexity, including passing of service configuration into this class.

Reply:
Pay attention @rhyslbw ,

I chosen an interface with optional dependencies in order to offer the caller the opportunity to use default values; otherwise the caller would be forced to always construct their selves a `CardanoTokenRegistry` and a `DbSyncNftMetadataService` which requires a `DbSyncMetadataService` causing a lot of code repetition.

For this reason I would prefer to not change the interface.

Reply:
> ...offer the caller the opportunity to use default values

I'm questioning this approach, as it's coupling the concrete services with this provider, rather than the interface only.

Reply:
Hi @rhyslbw  and @mkazlauskas ;

ca897f92f should solve this. Could you please check again?
Thank you

Reply:
Looks great!